tests/test_rate_limiter.py

```
@@ -0,0 +1,103 @@
+    mock_cursor.fetchone.side_effect = list(fetchone_values)
+    mock_cursor.rowcount = 0
+    mock_conn = MagicMock()
+    mock_conn.cursor.return_value.__enter__ = MagicMock(return_value=mock_cursor)
+    mock_conn.cursor.return_value.__exit__ = MagicMock(return_value=False)
+    return mock_conn, mock_cursor
+
+
+class TestCheckBurstLimit:
+
+    def test_allows_under_threshold(self):
+        conn, cursor = _make_conn([{"cnt": BURST_LIMIT - 1}])
```

Ruff (0.15.12)

[warning] Unpacked variable `cursor` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)